util/mutex.go

```diff
@@ -0,0 +1,108 @@
+package util
```

Comment:
copyright

Reply:
Ah yes, the thing I always manage to forget about.